scene/main/node.cpp

```diff
@@ -3528,6 +3541,9 @@ String Node::_get_name_num_separator() {
 
 Node::Node() {
 	orphan_node_count++;
+#ifdef TOOLS_ENABLED
+	connect(CoreStringNames::get_singleton()->script_changed, callable_mp(this, &Node::update_configuration_warnings));
+#endif
```

Comment:
I'm a bit wary of the cost of doing this for every single node instantiated. This should at least be limited to the editor only (`Engine::get_singleton()->is_editor_hint()`), but it would be good to benchmark too and make sure this doesn't have a measurable impact on the time to open the editor and open a complex scene with thousands of nodes.

Reply:
I don't think I would be able to benchmark this alone.

Reply:
I have been trying to compile a production build for this purpose but I cannot get it to work.

If I cannot get it to work, I would appreciate if someone were to try this in my place. I have attached a project with roughly 10.000 nodes, almost each of them with their own script, which may or may trigger the warning.
Although it's worth noting that most of the lag would probably come from the sheer amount of them, mainly.

[Node Script Warning Benchmark.zip](https://github.com/godotengine/godot/files/13967188/Node.Script.Warning.Benchmark.zip)

Reply:
Oh! I may have gotten actual numbers here!

After running both projects with these parameters:
```
hyperfine.exe --ignore-failure -w 3 -m 10 --parameter-list potato vanilla,node-warning
```

```
Benchmark 1: C:\Users\Mickeon\Documents\GitHub\godot\bin\vanilla\godot.windows.editor.x86_64.console.exe -e --quit-after 1 --path "C:\Users\Mickeon\Documents\Godot\Mini\Node Script Warning Benchmark"
  Time (mean ± σ):      3.818 s ±  0.009 s    [User: 3.233 s, System: 0.508 s]
  Range (min … max):    3.806 s …  3.834 s    10 runs

  Warning: Ignoring non-zero exit code.

Benchmark 2: C:\Users\Mickeon\Documents\GitHub\godot\bin\node-warning\godot.windows.editor.x86_64.console.exe -e --quit-after 1 --path "C:\Users\Mickeon\Documents\Godot\Mini\Node Script Warning Benchmark"
  Time (mean ± σ):      4.020 s ±  0.012 s    [User: 3.331 s, System: 0.500 s]
  Range (min … max):    3.992 s …  4.036 s    10 runs

  Warning: Ignoring non-zero exit code.

Summary
  C:\Users\Mickeon\Documents\GitHub\godot\bin\vanilla\godot.windows.editor.x86_64.console.exe -e --quit-after 1 --path "C:\Users\Mickeon\Documents\Godot\Mini\Node Script Warning Benchmark" ran
    1.05 ± 0.00 times faster than C:\Users\Mickeon\Documents\GitHub\godot\bin\node-warning\godot.windows.editor.x86_64.console.exe -e --quit-after 1 --path "C:\Users\Mickeon\Documents\Godot\Mini\Node Script Warning Benchmark"
```

Reply:
So yeah, assuming this test is reliable, this PR incurs a 5% performance drop in this case. That's somewhat significant.